From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Tue, 25 Aug 2026 20:05:04 +0200
Subject: fix(i18n): support find applet from busybox in module-setup.sh

busybox's `find` applet neither supports `-delete` nor specifying
multiple types to `-type`. So replace `-delete` by a `rm` call and split
the `-type` option.

Part of: https://github.com/dracut-ng/dracut/issues/2437
Origin: upstream, https://github.com/dracut-ng/dracut/commit/e241be2e0265616fbb32a8435eee4ee99cbb5bb1
---
 modules.d/20i18n/module-setup.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules.d/20i18n/module-setup.sh b/modules.d/20i18n/module-setup.sh
index a92939f..6e4f1d5 100755
--- a/modules.d/20i18n/module-setup.sh
+++ b/modules.d/20i18n/module-setup.sh
@@ -39,7 +39,7 @@ install() {
             MAPNAME=${1%.map*}
 
             mapfile -t -d '' MAPS < <(
-                find "${dracutsysrootdir-}${kbddir}"/keymaps/ -type f,l \( -name "${MAPNAME}" -o -name "${MAPNAME}.map*" \) -print0
+                find "${dracutsysrootdir-}${kbddir}"/keymaps/ \( -type f -o -type l \) \( -name "${MAPNAME}" -o -name "${MAPNAME}.map*" \) -print0
             )
         fi
 
@@ -159,7 +159,7 @@ install() {
 
         # remove unnecessary files
         rm -f -- "${initdir}${kbddir}/consoletrans/utflist"
-        find "${initdir}${kbddir}/" -name README\* -delete
+        find "${initdir}${kbddir}/" -name README\* -exec rm -f {} +
         find "${initdir}${kbddir}/" -name '*.gz' -print -quit \
             | while read -r _line || [ -n "$_line" ]; do
                 inst_multiple gzip
